### PR TITLE
Menu: convert Light DOM LESS to SCSS

### PR DESCRIPTION
**Pull request by BY00565233**
JIRA: SURF-1905

### LGTM's
* [x] Dev LGTM
* [x] Dev LGTM

### CODE REVIEW

File: docs/components/_index.scss [hunk 1]
```diff
@@ -1,4 +1,6 @@
 // load styles required to demonstrate each component
 
 @import "grid/index";
+@import "menu/spec";
```
Comment (CITguy) [inline docs/components/_index.scss]: To keep things a bit more organized, please place "spec" files in a new section below the "index" files (don't mix "spec" and "index" in the same group of imports).

```scss
/// load styles required to demonstrate each component
@import "grid/index";
@import "typography/index";

/// load styles needed to test component visuals
@import "menu/spec";
```

File: src/scss/components/menu/_index.scss [hunk 2]
```diff
@@ -1,26 +1,27 @@
...
-@import (reference) 'components/box/index';
-
-hx-menu:extend(.hxComponent) {
-  #Positionable.base();
-
-  background-color: @gray-0;
-  border: 1px solid @gray-300;
-  box-shadow: @layering-shadow;
+@import "mixins";
+@import "components/box/index";
```
[the remark this answers is not in the document]
Comment (BY00565233) [reply src/scss/components/menu/_index.scss]: Changes are done.

File: docs/components/_index.scss (hunk 1 above)
Comment (BY00565233) [reply docs/components/_index.scss]: Changes are done.

File: src/scss/components/menu/_index.scss [hunk 3]
```diff
@@ -32,18 +33,17 @@ hx-menu:extend(.hxComponent) {
   }
 }
 
-/*
-  Most of the styles mimic those of .hxBtn.hxLink.
+ // Most of the styles mimic those of .hxBtn.hxLink.
+
+ // However, because the generated CSS would end up being
+ // equally specific, extended definitions seem to take
+ // priority, so we can't use &:extend() here.
```
[the remark this answers is not in the document]
Comment (BY00565233) [reply src/scss/components/menu/_index.scss]: Changes are done.

File: src/scss/components/menu/_index.scss [hunk 4]
```diff
@@ -32,18 +33,17 @@ hx-menu:extend(.hxComponent) {
...
```
[the remark this answers is not in the document]
Comment (BY00565233) [reply src/scss/components/menu/_index.scss]: Changes are done.

**Pull request merged**